tests/Doctrine/Tests/DBAL/Functional/LikeWildcardsEscapingTest.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional;
+
+use Doctrine\DBAL\Platforms\OraclePlatform;
+use Doctrine\Tests\DbalFunctionalTestCase;
+
+final class LikeWildcardsEscapingTest extends DbalFunctionalTestCase
+{
+    public function testFetchLikeExpressionResult() : void
+    {
+        $string           = '_25% off_ your next purchase \o/';
+        $escapeChar       = '!';
+        $databasePlatform = $this->_conn->getDatabasePlatform();
+        $stmt             = $this->_conn->prepare(sprintf(
+            "SELECT (CASE WHEN '%s' LIKE '%s' ESCAPE '%s' THEN 1 ELSE 0 END)" .
+                ($databasePlatform instanceof OraclePlatform ? ' FROM dual' : ''),
```

Comment:
Personally, I use `$platform->getDummySelectSQL()` and replace the `1` with the needed expression for better portability.

@Ocramius would it make sense to add an optional parameter to `getDummySelectSQL()` which would accept an arbitrary SQL expression and use it instead of `1`? This might come in handy.

Reply:
Ping, @Ocramius.